app/utilities/supplementary_data_parser.py

```diff
@@ -0,0 +1,88 @@
+from typing import Mapping
+
+from marshmallow import (
+    EXCLUDE,
+    INCLUDE,
+    Schema,
+    ValidationError,
+    fields,
+    validate,
+    validates_schema,
+)
+
+from app.authentication.auth_payload_versions import SupplementaryDataSchemaVersion
+from app.utilities.metadata_parser_v2 import VALIDATORS, StripWhitespaceMixin
+
+
+class ItemsSchema(Schema):
+    identifier = VALIDATORS["string"](validate=validate.Length(min=1))
+
+
+class ItemsData(Schema, StripWhitespaceMixin):
+    pass
+
+
+class SupplementaryData(Schema, StripWhitespaceMixin):
+    identifier = VALIDATORS["string"](validate=validate.Length(min=1))
+    schema_version = VALIDATORS["string"](
+        validate=validate.OneOf([SupplementaryDataSchemaVersion.V1.value])
+    )
+    items = fields.Nested(ItemsData, required=False, unknown=EXCLUDE)
+
+    @validates_schema()
+    def validate_unit_id(self, data, **kwargs):
+        # pylint: disable=no-self-use, unused-argument
+        if (
+            data
+            and data.get("identifier")
+            and data["identifier"] != self.context["ru_ref"]
+        ):
+            raise ValidationError(
+                "Supplementary data did not return the specified Unit ID"
+            )
+
+
+class SupplementaryDataMetadataSchema(Schema, StripWhitespaceMixin):
```

Comment:
I wasn't imagining a schema for the API response itself, if we are doing this, this needs to handle other keys exists at the top level that we don't care about.

Reply:
It handles that with `Include` at the moment, `test_validate_supplementary_data_payload_with_unknown_field` was added to cover that if that's what you meant?

Reply:
All Include in this file now 👍